template-tarot.js

```diff
@@ -2,35 +2,62 @@
 var Card = (function(){
 
 	// Ctor:
-	function Card(id) {
-		//...
-	}
-
+	function Card(id) {			
+				this.id = id  //personal property			
+		}
 
 	// Instance methods (attach these to Card's prototype):
 
-	//rank...
-
-	//suit...
-
-	//name...
-
-	//color...
-
+	 Card.prototype.rank = function() { // --> 1..13, NaN
+        return Math.floor(this.id/4)+1;
+    }
+
+	
+    Card.prototype.suit = function() { // --> 1..4, NaN
+        return (this.id%4)+1;
+    }
+	
+	Card.prototype.name = function() { //--> string, NaN
+        var rankVal = this.rank();
+        var suitVal = this.suit();
+        return rankVal && suitVal &&
+            (this.constructor.rankNames()[rankVal-1]
+            +' of '						 		//-1 b/c no leading blank in arrays
+            +this.constructor.suitNames()[suitVal-1]);
+	}
 
+	
+	
+	Card.prototype.color = function() { // -->"red,"black",NaN
+		var suit=this.suit();
+		return suit && ((suit<3)? "red": "black");
+	}
+	
+	Card.prototype.isValid = function() { // Returns--> true, false
+        return ((typeof this.id)==="number") //correct type
+                && (this.id%1 === 0)        //integer
+                && this.id>=0 && this.id<=51;   //in range
```

Comment:
This is the source of one of your failing tests--a deck of tarot cards has more cards than a standard deck, so the upper end of the range needs to be different. There are several ways to do this, but probably the most graceful is to have a property in the prototype of each type of card that holds the number of cards in a deck, and use that as the upper end of the range.